starboard/android/shared/video_window.cc

```diff
@@ -72,6 +72,10 @@ void ClearNativeWindow(ANativeWindow* native_window) {
   // First, query how many configs match the given attribute list.
   EGLint num_configs = 0;
   EGL_CALL(eglChooseConfig(display, kAttributeList, NULL, 0, &num_configs));
+  if (num_configs == 0) {
+    SB_DLOG(ERROR) << "Found no configs in ClearVideoWindow";
+    return;
+  }
   SB_DCHECK(num_configs != 0);
```

Comment:
the stacktrace in the bug does not show which line is it from, could the crash caused by fail of the eglCreateContext() method below?

Reply:
The stack trace shows eglSwapBuffers() failed with an error. My thinking is that num_configs may have returned 0, then the `surface == EGL_NO_SURFACE` did not trigger an early return because `surface` is uninitialized in this case. Though I'm not clear whether the other EGL calls would trigger an error -- I imagine they should have. But at any rate, this is a safe enough change.

Reply:
Oh, I see, at the top, it says the crash is from line 119.
There are 2 SB_DCHECK after eglCreateContext(), if those fails, will eglSwapBuffers() crash as well? Would we return there?

Reply:
Yes, we can add early returns there as well. My thought is that since we're not seeing issues with those, let's keep the SB_DCHECKs since we aren't expecting them to fail. Then if they do fail, we can evaluate how best to handle the failures. Maybe we just shouldn't be calling this function on SbPlayer shutdown and clear the screen some other way (e.g. by going through the render_tree)?

Reply:
eglCreateContext() is followed by SB_DCHECK(eglGetError() == EGL_SUCCESS). If that failed, the crash would have been there instead of EGL_CALL(eglSwapBuffers(display, surface)). Similarly, eglChooseConfig is wrapped in EGL_CALL macro which includes a SB_DCHECK(eglGetError() == EGL_SUCCESS). So if that failed, then the crash log would not have pointed to that line.

Reply:
I thought SB_DCHECK does not crash for gold build. Oh, so the crash log is from a none-gold build?

The crash says it is from line 119, which is eglSwapBuffers(). On prod code, https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:starboard/android/shared/video_window.cc;l=119?q=p:cobalt%20starboard%2Fandroid%2Fshared%2Fvideo_window.cc, there are both SB_DCHECK after eglChooseConfig() and eglCreateContext(), how do you tell it is the one from the other?

Reply:
Yes, the bugreport from https://buganizer.corp.google.com/issues/277684908#comment26 shows that "versionName=3.04.010-FISHFOOD" was being used. So they actually had a devel build.